gdx/test/com/badlogic/gdx/utils/LongArrayTest.java

```diff
@@ -0,0 +1,425 @@
+		}
+	}
+
+	/** Test of the mul() method */
+	@Test
+	public void mulTest () {
+		LongArray longArray = new LongArray(new long[] {3, 4, 5, 1, 56, 32});
+		longArray.mul(1, 3);
+		Assert.assertEquals(12, longArray.get(1));
+		longArray.mul(2);
+		Assert.assertArrayEquals(new long[] {6, 24, 10, 2, 112, 64}, longArray.toArray());
+		try {
+			longArray.mul(17, 8);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// This should throw an exception as we are trying to multiply at an index out of bounds
+		}
+	}
+
+	/** Test of the insert() method */
+	@Test
+	public void insertTest () {
+		// With an ordered array
+		LongArray longArray1 = new LongArray();
+		longArray1.addAll(new long[] {1, 3, 4, 5, 6});
+		longArray1.insert(1, 2);
+		Assert.assertArrayEquals(new long[] {1, 2, 3, 4, 5, 6}, longArray1.toArray());
+		longArray1.insertRange(2, 3);
+		Assert.assertArrayEquals(new long[] {1, 2, 3, 4, 5, 3, 4, 5, 6}, longArray1.toArray());
+		try {
+			longArray1.insertRange(400, 4);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// This should throw an exception as we are trying to insert at an index out of bounds
+		}
+
+		// With an unordered array
+		LongArray longArray2 = new LongArray(false, 16);
+		longArray2.addAll(new long[] {1, 3, 4, 5, 6});
+		longArray2.insert(1, 2);
+		Assert.assertArrayEquals(new long[] {1, 2, 4, 5, 6, 3}, longArray2.toArray());
+		try {
+			longArray2.insert(2783, 3);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// This should throw an exception as we are trying to insert at an index out of bounds
+		}
+	}
+
+	/** Test of the swap() method */
+	@Test
+	public void swapTest () {
+		LongArray longArray1 = new LongArray(new long[] {1, 3, 4, 5, 6});
+		longArray1.swap(1, 4);
+		Assert.assertArrayEquals(new long[] {1, 6, 4, 5, 3}, longArray1.toArray());
+		try {
+			longArray1.swap(100, 3);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// This should throw an exception as we are trying to swap at an index out of bounds
+		}
+		try {
+			longArray1.swap(3, 100);
+			Assert.fail();
+		} catch (IndexOutOfBoundsException e) {
+			// This should throw an exception as we are trying to swap at an index out of bounds
+		}
+	}
+
+	@Test
+	public void containsTest () {
+		LongArray longArray1 = new LongArray(new long[] {1, 3, 4, 5, 6});
+		Assert.assertTrue(longArray1.contains(3));
+		Assert.assertFalse(longArray1.contains(100));
+	}
+
+	/** Test of the hashCode() method */
+	@Test
+	public void hashCodeTest () {
+		/*
+		 * Testing if the hash code of an ordered long array is in fact the same provided as the hashCode() method from Object.
+		 */
+		Object longArrayUnordered = new LongArray(false, 16);
+		((LongArray)longArrayUnordered).add(2, 1, -6, 2);
+		((LongArray)longArrayUnordered).add(5, 50, 34, 42);
+		Assert.assertEquals(longArrayUnordered.hashCode(), ((LongArray)longArrayUnordered).hashCode());
```

Comment:
This assertion is useless. `LongArray` implements `hashCode` - you are comparing the same result in each case. Try it - I replaced the `LongArray` implementation with `return 1` and this part still passed.

Reply:
Honestly any test on hashCode() is probably pointless. I say get rid of the hash code test.